### PR TITLE
#8683: Add Unary left shift

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Unary Left shift ( Issue #8683 )

- Test File : `tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_left_shift.py`
- CI : [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/runs/9380189231) - PASSED
- GS implementation is TODO, Hence test skipped for GS
- WH_BO Result :
Screenshot 2024-05-28 at 8 38 32 PM

- Documentation : 
Screenshot 2024-06-05 at 1 08 13 PM

### CODE REVIEW

File: tt_metal/hw/ckernels/grayskull/metal/llk_api/llk_sfpu/ckernel_sfpu_left_shift.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+//TODO: Implementation pending for GS
+template <bool APPROXIMATION_MODE, int ITERATIONS = 4>
+inline void calculate_left_shift(uint value) {
+#pragma GCC unroll 0
```
Comment (rtawfik01) [inline tt_metal/hw/ckernels/grayskull/metal/llk_api/llk_sfpu/ckernel_sfpu_left_shift.h:19]: Can you add `static_assert("GS not supported for this op");` and add the issue number?

File: tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_left_shift.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#include "ckernel_defs.h"
+#include "noc_nonblocking_api.h"
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+template <bool APPROXIMATION_MODE, int ITERATIONS = 8>
+inline void calculate_left_shift(uint value) {
```
Comment (rtawfik01) [inline tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_left_shift.h:17]: Can you change this to `const uint value`.

File: tt_metal/include/compute_kernel_api.h [hunk 3]
```diff
@@ -862,6 +862,30 @@ ALWI void unary_gt_tile_init() {
     MATH(( llk_math_eltwise_unary_sfpu_unary_gt_init<APPROX>() ));
 }
 
+/**
+ * Performs element-wise left_shift computation on input x , where x is each element of a tile
```
Comment (rtawfik01) [inline tt_metal/include/compute_kernel_api.h:866]: Update description to say this operation only works with integer datatypes

File: tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_xary_ops.cpp [hunk 4]
```diff
@@ -152,6 +152,12 @@ namespace tt::tt_metal::detail {
             R"doc("value", "float", "")doc"
 
         );
+        detail::bind_unary_op_with_param(
+            m_tensor, "left_shift", left_shift,
+            py::arg("value"),
+            R"doc(Computes left shift of input tensor ``{0}`` by ``{1}`` bits.)doc",
+            R"doc("value", "uint", "")doc"
+        );
```
Comment (tt-aho) [inline tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_xary_ops.cpp:160]: Add line that says it is not supported on GS

**Review by rtawfik01 — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Pull request merged**